src/murmur/Server.cpp

```diff
@@ -501,6 +507,21 @@ void Server::readParams() {
 	}
 }
 
+void Server::loadWelcomeTextFromFile() {
+	if (qsWelcomeText.isEmpty()) {
+		QFile f(qsWelcomeTextFile);
+		if (f.open(QFile::ReadOnly | QFile::Text)) {
+			QTextStream in(&f);
+			qsWelcomeText = in.readAll();
+			f.close();
+		} else {
+			log(QString("Failed to open welcome text file %1").arg(qsWelcomeTextFile));
+		}
+	} else {
+		log(QString("Ignoring welcometextfile %1 because welcometext is defined").arg(qsWelcomeTextFile));
+	}
+}
```

Comment:
Maybe this should return a QString instead of setting the value directly?

Reply:
Sounds like a good idea, yes :+1: 

The check whether the `qsWelcomeText` is empty could then also be done before calling this function.